File: tests/agent_workloads/adapters/structured_data.py
import json
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: structured_data INPUT_FILE")

try:
    payload = json.loads(Path(sys.argv[1]).read_text())
    events = payload["events"]
    if not isinstance(events, list):
        raise ValueError
    for event in events:
        if not isinstance(event, dict):
            raise ValueError
        if not isinstance(event["service"], str) or not isinstance(event["duration_ms"], int):
            raise ValueError
except (KeyError, TypeError, ValueError, json.JSONDecodeError):
    print("invalid-json")
else:
    summaries = []
    for service in sorted({event["service"] for event in events}):
        rows = [event for event in events if event["service"] == service]
        summaries.append(
            {"service": service, "count": len(rows), "total_ms": sum(row["duration_ms"] for row in rows)}
        )
    print(json.dumps({"total_events": len(events), "summaries": summaries}, separators=(",", ":")))

File: tests/agent_workloads/adapters/structured_data.mjs
import { readFileSync } from "node:fs";

if (process.argv.length !== 3) {
  throw new Error("usage: structured_data INPUT_FILE");
}

try {
  const payload = JSON.parse(readFileSync(process.argv[2], "utf8"));
  if (!Array.isArray(payload.events)) throw new Error("events");
  for (const event of payload.events) {
    if (
      event === null ||
      typeof event.service !== "string" ||
      !Number.isInteger(event.duration_ms)
    ) throw new Error("event");
  }
  const summaries = [...new Set(payload.events.map((event) => event.service))]
    .sort()
    .map((service) => {
      const rows = payload.events.filter((event) => event.service === service);
      return {
        service,
        count: rows.length,
        total_ms: rows.reduce((total, event) => total + event.duration_ms, 0),
      };
    });
  console.log(JSON.stringify({ total_events: payload.events.length, summaries }));
} catch {
  console.log("invalid-json");
}
